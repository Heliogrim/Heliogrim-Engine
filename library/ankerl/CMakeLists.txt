function(library_ankerl_init)

	cmake_minimum_required(VERSION 3.27.7)
	include(FetchContent)

	name_to_external_alias(alias "ankerl")

	# Include Sub-Project
	get_external_include_path(lib_inc_path "ankerl")
	get_external_source_path(lib_src_path "ankerl")

	FetchContent_Declare(
			ankerl
			GIT_REPOSITORY https://github.com/martinus/unordered_dense.git
			GIT_TAG v4.4.0
			GIT_SUBMODULES_RECURSE OFF
			GIT_SHALLOW ON
			SOURCE_DIR "${lib_src_path}"
			PATCH_COMMAND ${CMAKE_COMMAND} -E rm -f -- "${lib_src_path}/CMakeLists.txt"
			LOG_DOWNLOAD ON
			EXCLUDE_FROM_ALL
	)
	FetchContent_MakeAvailable(
			ankerl
	)

	# Setup Library
	install_includes(
			DIRECTORY "${lib_src_path}/include/"
			DESTINATION "${lib_inc_path}/"
	)

endfunction()

# Push to setup

add_managed_library(NAME ankerl CALL library_ankerl_init ORDER_INDEPENDENT ON)