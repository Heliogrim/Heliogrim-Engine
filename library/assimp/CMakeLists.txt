function(library_assimp_init)

	cmake_minimum_required(VERSION 3.27.7)
	include(FetchContent)

	name_to_external_alias(alias "assimp")

	#
	set(ASSIMP_HUNTER_ENABLED OFF CACHE INTERNAL "" FORCE)
	set(BUILD_SHARED_LIBS ON CACHE INTERNAL "" FORCE)
	set(ASSIMP_BUILD_FRAMEWORK OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_DOUBLE_PRECISION OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_OPT_BUILD_PACKAGES OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_ANDROID_JNIIOSYSTEM OFF CACHE INTERNAL "" FORCE)

	set(ASSIMP_NO_EXPORT ON CACHE INTERNAL "" FORCE)
	set(ASSIMP_INSTALL OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_INSTALL_PDB OFF CACHE INTERNAL "" FORCE)

	set(ASSIMP_BUILD_ASSIMP_TOOLS OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_BUILD_SAMPLES OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_BUILD_TESTS OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_COVERALLS OFF CACHE INTERNAL "" FORCE)

	set(ASSIMP_WARNINGS_AS_ERRORS OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_ASAN OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_UBSAN OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_BUILD_DOCS OFF CACHE INTERNAL "" FORCE)
	set(ASSIMP_BUILD_ZLIB OFF CACHE INTERNAL "" FORCE)
	set(USE_STATIC_CRT OFF CACHE INTERNAL "" FORCE)

	get_external_include_path(zlib_include_path "zlib")
	set(ZLIB_INCLUDE_DIR "${zlib_include_path}/zlib" CACHE INTERNAL "" FORCE)
	set(ZLIB_LIBRARY ${PROJECT_NAME}::External-zlib CACHE INTERNAL "" FORCE)
	set(ZLIB_LIBRARIES ${PROJECT_NAME}::External-zlib CACHE INTERNAL "" FORCE)
	set(ZLIB_FOUND ON CACHE INTERNAL "" FORCE)

	# Include Sub-Project
	get_library_path(lib_base_path)
	get_external_include_path(lib_inc_path "assimp")
	get_external_source_path(lib_src_path "assimp")

	FetchContent_Declare(
			libassimp
			GIT_REPOSITORY https://github.com/assimp/assimp.git
			GIT_TAG v5.4.0
			GIT_SHALLOW ON
			GIT_PROGRESS ON
			SOURCE_DIR "${lib_src_path}"
			LOG_DOWNLOAD ON
			EXCLUDE_FROM_ALL
	)
	FetchContent_MakeAvailable(
			libassimp
	)

	# Setup Library
	add_library(${PROJECT_NAME}::${alias} ALIAS assimp)

	# Setup Library
	install_includes(
			DIRECTORY "${lib_src_path}/include/"
			DESTINATION "${lib_inc_path}/"
	)

	export(TARGETS assimp NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${alias}/${alias}-export.cmake)

	# Deployment
	get_external_lib_path(lib_dist_path "assimp")
	set_batch_target_properties(assimp ${lib_dist_path})

	# Reset Settings
	set(BUILD_SHARED_LIBS OFF CACHE INTERNAL "" FORCE)

endfunction()

# Push to setup
add_managed_library(NAME libassimp CALL library_assimp_init DEPS libzlib)