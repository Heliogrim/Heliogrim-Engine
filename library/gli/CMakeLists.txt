function(library_gli_init)

	cmake_minimum_required(VERSION 3.27.7)
	include(FetchContent)

	name_to_external_alias(alias "gli")

	# Include Sub-Project
	get_external_include_path(lib_inc_path "gli")
	get_external_source_path(lib_src_path "gli")

	FetchContent_Declare(
			gli
			GIT_REPOSITORY https://github.com/g-truc/gli.git
			GIT_TAG master
			GIT_SUBMODULES_RECURSE OFF
			GIT_SHALLOW ON
			SOURCE_DIR "${lib_src_path}"
			PATCH_COMMAND ${CMAKE_COMMAND} -E rm -f -- "${lib_src_path}/CMakeLists.txt"
			LOG_DOWNLOAD ON
			EXCLUDE_FROM_ALL
	)
	FetchContent_MakeAvailable(
			gli
	)

	# Setup Library
	install_includes(
			DIRECTORY "${lib_src_path}/gli/"
			DESTINATION "${lib_inc_path}/gli/"
	)

endfunction()

# Push to setup
add_managed_library(NAME gli CALL library_gli_init ORDER_INDEPENDENT ON)