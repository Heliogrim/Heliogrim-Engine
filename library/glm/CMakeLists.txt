function(library_glm_init)

	cmake_minimum_required(VERSION 3.27.7)
	include(FetchContent)

	name_to_external_alias(alias "glm")

	# Include Sub-Project
	get_external_include_path(lib_inc_path "glm")
	get_external_source_path(lib_src_path "glm")

	FetchContent_Declare(
			glm
			GIT_REPOSITORY https://github.com/g-truc/glm.git
			GIT_TAG 1.0.1
			GIT_SUBMODULES_RECURSE OFF
			GIT_SHALLOW ON
			SOURCE_DIR "${lib_src_path}"
			PATCH_COMMAND ${CMAKE_COMMAND} -E rm -f -- "${lib_src_path}/CMakeLists.txt"
			CONFIGURE_COMMAND ""
			LOG_DOWNLOAD ON
			EXCLUDE_FROM_ALL
	)
	FetchContent_MakeAvailable(
			glm
	)

	# Setup Library
	install_includes(
			DIRECTORY "${lib_src_path}/glm/"
			DESTINATION "${lib_inc_path}/glm/"
	)

endfunction()

# Push to setup
add_managed_library(NAME glm CALL library_glm_init ORDER_INDEPENDENT ON)