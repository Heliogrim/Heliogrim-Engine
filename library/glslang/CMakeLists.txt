function(library_glslang_init)

	cmake_minimum_required(VERSION 3.27.7)
	include(FetchContent)

	name_to_external_alias(glslang_alias "glslang")
	name_to_external_alias(generic_code_gen_alias "GenericCodeGen")
	name_to_external_alias(machine_indep_alias "MachineIndependent")
	name_to_external_alias(os_dep_alias "OSDependent")
	name_to_external_alias(default_limits_alias "glslang-default-resource-limits")
	name_to_external_alias(spirv_alias "SPIRV")
	name_to_external_alias(spv_remapper_alias "SPVRemapper")

	#
	set(BUILD_SHARED_LIBS OFF CACHE INTERNAL "" FORCE)
	set(BUILD_EXTERNAL OFF CACHE INTERNAL "" FORCE)
	set(GLSLANG_TESTS OFF CACHE INTERNAL "" FORCE)
	set(GLSLANG_TESTS_DEFAULT OFF CACHE INTERNAL "" FORCE)
	set(GLSLANG_ENABLE_INSTALL OFF CACHE INTERNAL "" FORCE)
	set(ENABLE_GLSLANG_JS OFF CACHE INTERNAL "" FORCE)

	set(ENABLE_SPVREMAPPER ON CACHE INTERNAL "" FORCE)
	set(ENABLE_GLSLANG_BINARIES OFF CACHE INTERNAL "" FORCE)

	set(ENABLE_HLSL ON CACHE INTERNAL "" FORCE)
	set(ENABLE_RTTI OFF CACHE INTERNAL "" FORCE)
	set(ENABLE_EXCEPTIONS ON CACHE INTERNAL "" FORCE)

	set(ENABLE_OPT ON CACHE INTERNAL "" FORCE)
	set(ALLOW_EXTERNAL_SPIRV_TOOLS OFF CACHE INTERNAL "" FORCE)

	# Include Sub-Project
	get_external_include_path(lib_inc_path "glslang")
	get_external_source_path(lib_src_path "glslang")

	FetchContent_Declare(
			libglslang
			GIT_REPOSITORY https://github.com/KhronosGroup/glslang.git
			GIT_TAG 14.1.0
			GIT_SHALLOW ON
			GIT_PROGRESS ON
			SOURCE_DIR "${lib_src_path}"
	)
	FetchContent_MakeAvailable(
			libglslang
	)

	# Setup Library
	add_library(${PROJECT_NAME}::${glslang_alias} ALIAS glslang)
	add_library(${PROJECT_NAME}::${generic_code_gen_alias} ALIAS GenericCodeGen)
	add_library(${PROJECT_NAME}::${machine_indep_alias} ALIAS MachineIndependent)
	add_library(${PROJECT_NAME}::${os_dep_alias} ALIAS OSDependent)
	add_library(${PROJECT_NAME}::${default_limits_alias} ALIAS glslang-default-resource-limits)
	add_library(${PROJECT_NAME}::${spirv_alias} ALIAS SPIRV)
	add_library(${PROJECT_NAME}::${spv_remapper_alias} ALIAS SPVRemapper)

	install_includes(DIRECTORY "${lib_src_path}/glslang/HLSL/" DESTINATION "${lib_inc_path}/glslang/HLSL/")
	install_includes(DIRECTORY "${lib_src_path}/glslang/Include/" DESTINATION "${lib_inc_path}/glslang/Include/")
	install_includes(DIRECTORY "${lib_src_path}/glslang/MachineIndependent/" DESTINATION "${lib_inc_path}/glslang/MachineIndependent/")
	install_includes(DIRECTORY "${lib_src_path}/glslang/Public/" DESTINATION "${lib_inc_path}/glslang/Public/")
	install_includes(DIRECTORY "${lib_src_path}/SPIRV/" DESTINATION "${lib_inc_path}/glslang/SPIRV/")

	export(TARGETS glslang NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${glslang_alias}/${glslang_alias}-export.cmake)
	export(TARGETS GenericCodeGen NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${generic_code_gen_alias}/${generic_code_gen_alias}-export.cmake)
	export(TARGETS MachineIndependent NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${machine_indep_alias}/${machine_indep_alias}-export.cmake)
	export(TARGETS OSDependent NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${os_dep_alias}/${os_dep_alias}-export.cmake)
	export(TARGETS glslang-default-resource-limits NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${default_limits_alias}/${default_limits_alias}-export.cmake)
	export(TARGETS SPIRV NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${spirv_alias}/${spirv_alias}-export.cmake)
	export(TARGETS SPVRemapper NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${spv_remapper_alias}/${spv_remapper_alias}-export.cmake)

	# Deployment
	get_external_lib_path(lib_dist_path "glslang")
	foreach (target glslang GenericCodeGen MachineIndependent OSDependent glslang-default-resource-limits SPIRV SPVRemapper)
		set_batch_target_properties(${target} ${lib_dist_path})
	endforeach ()

endfunction()

# Push to setup
add_managed_library(NAME libglslang CALL library_glslang_init DEPS libspirv-tools libspirv-headers)