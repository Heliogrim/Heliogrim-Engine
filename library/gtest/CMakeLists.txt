function(library_gtest_init)

	cmake_minimum_required(VERSION 3.27.7)
	include(FetchContent)

	name_to_external_alias(gtest_alias "gtest")
	name_to_external_alias(gtest_main_alias "gtest_main")
	name_to_external_alias(gmock_alias "gmock")
	name_to_external_alias(gmock_main_alias "gmock_main")

	#
	set(BUILD_SHARED_LIBS OFF CACHE INTERNAL "" FORCE)

	set(BUILD_GMOCK ON CACHE INTERNAL "" FORCE)
	set(INSTALL_GTEST OFF CACHE INTERNAL "" FORCE)
	set(GTEST_HAS_ABSL OFF CACHE INTERNAL "" FORCE)

	set(gtest_force_shared_crt OFF CACHE INTERNAL "" FORCE)
	set(gtest_build_tests OFF CACHE INTERNAL "" FORCE)
	set(gtest_build_samples OFF CACHE INTERNAL "" FORCE)
	set(gtest_disable_pthreads OFF CACHE INTERNAL "" FORCE)
	set(gtest_hide_internal_symbols OFF CACHE INTERNAL "" FORCE)

	set(gmock_build_tests OFF CACHE INTERNAL "" FORCE)

	# Include Sub-Project
	get_external_source_path(lib_src_path "gtest")
	get_external_include_path(lib_inc_path "gtest")

	FetchContent_Declare(
			googletest
			GIT_REPOSITORY https://github.com/google/googletest.git
			GIT_TAG v1.14.0
			GIT_SHALLOW ON
			SOURCE_DIR "${lib_src_path}"
			EXCLUDE_FROM_ALL
	)
	FetchContent_MakeAvailable(
			googletest
	)

	# Setup Library
	add_library(${PROJECT_NAME}::${gtest_alias} ALIAS gtest)
	add_library(${PROJECT_NAME}::${gtest_main_alias} ALIAS gtest_main)
	add_library(${PROJECT_NAME}::${gmock_alias} ALIAS gmock)
	add_library(${PROJECT_NAME}::${gmock_main_alias} ALIAS gmock_main)

	install_includes(
			DIRECTORY "${lib_src_path}/googletest/include/" "${lib_src_path}/googlemock/include/"
			DESTINATION "${lib_inc_path}/"
	)

	export(TARGETS gtest NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${gtest_alias}/${gtest_alias}-export.cmake)
	export(TARGETS gtest_main NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${gtest_main_alias}/${gtest_main_alias}-export.cmake)
	export(TARGETS gmock NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${gmock_alias}/${gmock_alias}-export.cmake)
	export(TARGETS gmock_main NAMESPACE External:: FILE ${PROJECT_BINARY_DIR}/cmake/${gmock_main_alias}/${gmock_main_alias}-export.cmake)

	# Deployment
	get_external_lib_path(lib_dist_path "gtest")
	foreach (target gtest gtest_main gmock gmock_main)
		set_batch_target_properties(${target} ${lib_dist_path})
	endforeach ()

endfunction()

# Push to setup
add_managed_library(
		NAME googletest
		CALL library_gtest_init
		ORDER_INDEPENDENT ON
		TARGETS gtest gtest_main gmock gmock_main
)