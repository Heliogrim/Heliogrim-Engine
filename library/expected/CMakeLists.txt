function(library_expected_init)

	cmake_minimum_required(VERSION 3.27.7)
	include(FetchContent)

	name_to_external_alias(alias "expected")

	# Include Sub-Project
	get_external_include_path(lib_inc_path "expected")
	get_external_source_path(lib_src_path "expected")

	FetchContent_Declare(
			expected
			GIT_REPOSITORY https://github.com/TartanLlama/expected.git
			GIT_TAG v1.1.0
			GIT_SUBMODULES_RECURSE OFF
			GIT_SHALLOW ON
			SOURCE_DIR "${lib_src_path}"
			PATCH_COMMAND ${CMAKE_COMMAND} -E rm -f -- "${lib_src_path}/CMakeLists.txt"
			LOG_DOWNLOAD ON
			EXCLUDE_FROM_ALL
	)
	FetchContent_MakeAvailable(
			expected
	)

	# Setup Library
	install_includes(
			DIRECTORY "${lib_src_path}/include/"
			DESTINATION "${lib_inc_path}/"
	)

endfunction()

# Push to setup
add_managed_library(NAME expected CALL library_expected_init ORDER_INDEPENDENT ON)